File: src/test/java/pl/mjedynak/idea/plugins/builder/psi/MethodNameCreatorTest.java
package pl.mjedynak.idea.plugins.builder.psi;

import static org.apache.commons.lang3.StringUtils.EMPTY;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

public class MethodNameCreatorTest {

    @Test
    void shouldCreateMethodIfPrefixIsEmpty() {
        // when
        String result = MethodNameCreator.createMethodName(EMPTY, "userName");

        // then
        assertThat(result).isEqualTo("userName");
    }

    @Test
    void shouldCreateMethodWithCapitalizedFieldNameIfPrefixIsNotEmpty() {
        // when
        String result = MethodNameCreator.createMethodName("with", "field");

        // then
        assertThat(result).isEqualTo("withField");
    }
}

File: src/test/kotlin/io/github/kingg22/kobra/builder/psi/MethodNameCreatorTest.kt
package io.github.kingg22.kobra.builder.psi

import io.github.kingg22.kobra.builder.psi.MethodNameCreator.createMethodName
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MethodNameCreatorTest {
    @Test
    fun shouldCreateMethodIfPrefixIsEmpty() {
        // when
        val result = createMethodName("", "userName")

        // then
        assertThat(result).isEqualTo("userName")
    }

    @Test
    fun shouldCreateMethodWithCapitalizedFieldNameIfPrefixIsNotEmpty() {
        // when
        val result = createMethodName("with", "field")

        // then
        assertThat(result).isEqualTo("withField")
    }
}
